### PR TITLE
Offer variables fall back to product values when no discount phase

### PR DESCRIPTION
**Pull request by vegaro**
## Summary
- When `product.offer_*` variables have no discount available (no promo nor intro), they now fall back to their product equivalents instead of returning empty strings
- This prevents developers wrongly using `offer_price` in the default text from having an empty text

Android counterpart of iOS PR [#6242 ](https://github.com/RevenueCat/purchases-ios/pull/6242) (PW-160).
Closes PW-161.

**WARNING: this PR is easier to review by commit. I thought it was about time that we refactored that massive class and I extracted some of the code into their own files**

---

> [!NOTE]
> **Medium Risk**
> Changes paywall variable resolution for `product.offer_*` to return product-derived values instead of empty strings when no discount phase (or when a discount phase can’t be expressed in the requested unit), which could alter displayed pricing text in production paywalls.
> 
> **Overview**
> **Offer-related paywall variables now gracefully fall back to product values.** When `product.offer_*` variables have no applicable discount phase (no promo/intro), `VariableProcessorV2` resolves them using the corresponding product price/period instead of returning an empty string; `offer_price_per_*` also falls back when the discount phase `canDisplay` check fails for the requested unit.
> 
> As part of the change, period/offer computation helpers were refactored out of `VariableProcessorV2` into new extension files (`PackagePeriodExtensions`, `PeriodExtensions`, `PricingPhaseExtensions`), and tests were updated/expanded to cover the new fallback behavior (including secondary offers and can-display edge cases).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2c2675d79f12b371697199532f9c7023157a3da9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt [hunk 1]
```diff
@@ -477,64 +471,87 @@ internal object VariableProcessorV2 {
...
+            primaryDiscountPhase(subscriptionOption, rcPackage)?.productOfferPeriodInWeeks
+                ?: rcPackage?.product?.period?.roundedValueInWeeks
+
+        Variable.PRODUCT_OFFER_PERIOD_IN_MONTHS ->
+            primaryDiscountPhase(subscriptionOption, rcPackage)?.productOfferPeriodInMonths
+                ?: rcPackage?.product?.period?.roundedValueInMonths
+
+        Variable.PRODUCT_OFFER_PERIOD_IN_YEARS ->
+            primaryDiscountPhase(subscriptionOption, rcPackage)?.productOfferPeriodInYears
+                ?: rcPackage?.product?.period?.roundedValueInYears
```
Comment (facumenzella) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorV2.kt:542]: Minor nit: Primary offer variables use direct `?:` fallback (e.g., `PRODUCT_OFFER_PRICE` at line 471), but secondary offer variables use the `resolveOfferPrice`/`resolveOfferPeriod` helpers here. Consider using the helpers consistently for both primary and secondary, or inline both - it would make the code easier to follow.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/PricingPhaseExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+internal fun resolveOfferPrice(
+    discountPhase: PricingPhase?,
+    localizedVariableKeys: Map<VariableLocalizationKey, String>,
+    productFallback: () -> String?,
+): String? =
+    discountPhase?.productOfferPrice(localizedVariableKeys) ?: productFallback()
+
+internal fun resolveOfferPricePerPeriod(
+    discountPhase: PricingPhase?,
+    localizedVariableKeys: Map<VariableLocalizationKey, String>,
```
Comment (facumenzella) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/PricingPhaseExtensions.kt]: This helper function is defined but doesn't appear to be called anywhere. Is this dead code or intended for future use? Consider removing if unused.
